tests/model_registry/conftest.py

```diff
@@ -0,0 +1,307 @@
+import pytest
+from typing import Generator, Any
+from ocp_resources.secret import Secret
+from ocp_resources.namespace import Namespace
+from ocp_resources.service import Service
+from ocp_resources.persistent_volume_claim import PersistentVolumeClaim
+from ocp_resources.deployment import Deployment
+from ocp_resources.model_registry import ModelRegistry
+from simple_logger.logger import get_logger
+from kubernetes.dynamic import DynamicClient
+
+from tests.model_registry.utils import get_endpoint_from_mr_service, get_mr_service_by_label
+from utilities.infra import create_ns
+from utilities.constants import Protocols
+
+
+LOGGER = get_logger(name=__name__)
+
+DB_RESOURCES_NAME = "model-registry-db"
+
+
+@pytest.fixture(scope="class")
+def model_registry_namespace(admin_client: DynamicClient) -> Generator[Namespace, Any, Any]:
+    # This namespace should exist after Model Registry is enabled, but it can also be deleted
+    # from the cluster and does not get reconciled. Fetch if it exists, create otherwise.
+    ns = Namespace(name="rhoai-model-registries", client=admin_client)
+    if ns.exists:
+        yield ns
+    else:
+        with create_ns(
```

Comment:
I think logging warning could be helpful here. If it is already deleted and we are recreating it